### PR TITLE
Debtor data validation

### PR DESCRIPTION
**Pull request by gh2os**

### CODE REVIEW

**Comment by bryan-gilbert** [comment]
Please squash all those commits before merging. Thanks

File: ppr-api/src/schemas/party.py [hunk 1]
```diff
@@ -30,6 +30,14 @@ class Party(pydantic.BaseModel):
     businessName: str = None
     address: Address = None
 
+    @pydantic.validator('address', pre=None, always=True)
+    def check_party_for_business_name(cls, v, values, **kwargs):  # pylint:disable=no-self-argument # noqa: N805
+        if v is None:
+            raise ValueError('"Party" must have an address')
+        elif values['personName'] is None and values['businessName'] is None:
```
Comment (bryan-gilbert) [inline ppr-api/src/schemas/party.py:37]: Is there another place that person name and business name are tested for content?

File: ppr-api/tests/unit/endpoints/test_financing_statement.py [hunk 2]
```diff
@@ -135,7 +139,8 @@ def test_read_financing_statement_debtor_should_be_mapped_to_schema():
     assert debtor.personName.last == 'Simpson'
     assert debtor.businessName == 'Mr. Plow'
     assert debtor.birthdate == datetime.date(1990, 6, 15)
-    assert debtor.address is None
+    assert debtor.address == {'street': '123 Fake Street', 'city': 'Victoria', 'country': 'CA',
+                              'postalCode': 'V1V 1V1', 'streetAdditional': 'Suite 100', 'region': 'BC'}
```
Comment (bryan-gilbert) [inline ppr-api/tests/unit/endpoints/test_financing_statement.py:143]: Can we make these test more DRY?  If there is a long complex object that is used many times could it be constructed once?

File: ppr-api/tests/unit/endpoints/test_search.py [hunk 3]
```diff
@@ -122,21 +122,30 @@ def test_read_search_results_provides_secured_party_at_time_of_matched_registrat
     third_event = stub_financing_statement_event('123458C', financing_statement=fin_stmt)
     third_event.registration_date = now + datetime.timedelta(seconds=2)
     first_secured_party = models.party.Party(
-        type_code='SP', last_name='Simpson', first_name='Homer', base_registration_number=fin_stmt.registration_number,
+        type_code='SP', last_name='Simpson', first_name='Homer',  base_registration_number=fin_stmt.registration_number,
```
Comment (bryan-gilbert) [inline ppr-api/tests/unit/endpoints/test_search.py]: Did you intend on a double space before base_registration_number?

File: ppr-api/src/schemas/party.py (hunk 1 above)
Comment (bryan-gilbert) [reply ppr-api/src/schemas/party.py:37]: Forget that comment. I see the answer is below.

**Review by bryan-gilbert — APPROVED** [review]

**Comment by jguertin** [comment]
Closing, as it will not be completed

**Pull request closed without merging**